packages/fluentui/react-builder/src/components/Designer.tsx

```diff
@@ -166,6 +168,12 @@ const stateReducer: Reducer<DesignerState, DesignerAction> = (draftState, action
       treeChanged = true;
       break;
 
+    case 'UPLOAD_TREE':
+      draftState.jsonTree = action.jsonTree;
```

Comment:
@jurokapsiar @miroslavstastny Question for all the state change: Shouldn't we be returning a new state instead of changing it directly?

```
return {...draftState, jsonTre: action.jsonTree}
```

Reply:
looking at https://www.smashingmagazine.com/2020/06/better-reducers-with-immer/, they do both - sometines construct new state, sometimes mutate the old one...

Reply:
always mutate the draft state and return nothing. that's the purpose of immer.